add tutorial setup script to share/spack

becker33 opened the pull request:
Adds a script that helps setup an ubuntu18.04 machine that is not our tutorial container to run the Spack tutorial.

becker33 committed:
add tutorial setup script to share/spack

tgamblin commented on share/spack/setup-tutorial-env.sh:
```diff
@@ -0,0 +1,102 @@
+#     when we run on aws cloud9 we have to expand the storage from 10G to 30G
+#     because we install too much software for a default cloud9 instance
+###############################################################################
+
```
Suggest adding:

```bash
if [ -f /etc/os-release ]; then
    . /etc/os-release
fi
if [ x"$UBUNTU_CODENAME" != "xbionic" ]; then
    echo "The tutorial setup script must be run on Ubuntu 18.04."
    return 1
fi
```

tgamblin commented on share/spack/setup-tutorial-env.sh:
```diff
@@ -0,0 +1,102 @@
+####
+# AWS set volume size to 30G
+####
+
```
can probably exit here if:

```bash
if [ ! -f /sys/hypervisor/uuid ]; then
    return
fi
```

This tells you it's an EC2 instance.  We can probably do better eventually but for now this is good.

tgamblin committed:
Add check for Ubuntu 18, fix xvda check, fix apt-get errors

- now works on t2.micro, t2.small, and m instances
- apt-get needs retries around it to work

tgamblin reviewed (approved):
I had to add a few things to get this working reliably:

1. Retry loops around `apt-get` commands. Unattended updates seem to run immediately after you get you instance, which means that `apt-get` will complain about the lock being taken.  I just put a `sleep 5` until success loop around the commands.

2. I had to check for `[ -e /dev/xvda1 ]` to get the right results for diferrent instances.

3. I added a guard at the beginning to ensure we're on ubuntu.  The script will complain if it is on some other OS.
